machine_learning/create_dataset.py

```diff
@@ -0,0 +1,73 @@
+import os
+import argparse
+import web_monitoring.differs as wd
+import pandas as pd
+
+def text_diff(a_text, b_text):
+    """
+    Gets text changes between two versions.
+    Returns all changes of a certain category joined into a single string
+
+    Parameters
+    ----------
+    a_text, b_text* : string
+
+    Returns
+    -------
+    deleted : All deletions joined to form a single string
+    added : All additions joined to form a single string
+    combined : All changes joined to form a single string
+    len(diffs) : Total no. of changes computed by html_text_diff
+    """
+    diffs = wd.html_text_diff(a_text=a_text, b_text=b_text)
+    state = []
+    deleted = ' '.join([diff[1] for diff in diffs if diff[0] == -1])
+    added = ' '.join([diff[1] for diff in diffs if diff[0] == 1])
+    combined = ' '.join([diff[1] for diff in diffs])
+    return deleted, added, combined, len(diffs)
+
+def create_dataset(dirname):
+    """
+    Creates dataset for classification model
+
+    Parameters
+    ----------
+    dirname* : Name of the directory where all the versions(to be diffed)
+              are stored
+
+    Returns
+    -------
+    df : Pandas DataFrame with information about changes between
+                versions as rows
+    """
+    path = os.path.join(os.getcwd(), dirname)
+    filelist = os.listdir(path)
+    os.chdir(path)
+    columns = ['deleted', 'added', 'combined', 'no. of changes']
+    df = pd.DataFrame(columns=columns)
+    for index in range(0,len(filelist),2):
```

Comment:
Trivial style niggle: I think we're following the convention of putting spaces after commas. I'll mention just this one case, but it might be best to search the files for other possible cases. A very low priority item, though.